### ISSUE TITLE
Audio cuts out every ~2 min

### ISSUE THREAD

**Issue by bma-diy** [popcornmix/omxplayer#788]
I made a python script that plays a video at the push of a button using Omxplayer. It works fine most of the time, but occasionally it will have a problem where it cuts out audio for 3-4 seconds every 1 minute 56 seconds. I have been unable to reproduce it at will because a reboot seems to clear the issue. The issue seems to present after the pi has been on for a while.

I've tried different methods of using the omxplayer in python, the issue happened in both implementations:

- https://python-omxplayer-wrapper.readthedocs.io/en/latest/

- This has also happened when using the subprocess library to execute the omxplayer.

I've had this happen on two different Pi's: Raspberry Pi 3 B and my current Pi: Raspberry pi 3A+. The way I seemed to have fixed it the first time was by reflashing my Raspberry Pi and configuring everything from scratch.

This happens on my older Insignia/ Roku TV but not on my Asus monitor with speakers connected to it.

I've done all the basic stuff:

hdmi_force_hotplug=1
hdmi_group=1
hdmi_mode=16
hdmi_force_edid_audio=1
hdmi_drive=2
config_hdmi_boost=7 <- this did not help so I reverted back

The logs from /var/log don't tell me anything, and my python script doesn't throw any errors.

My question is: how can I properly troubleshoot omxplayer audio issues the next time it occurs?

**Comment by popcornmix** [comment]
omxplayer has been deprecated in favour of vlc. Do you get the same issue with vlc?

**Comment by bma-diy** [comment]
> omxplayer has been deprecated in favour of vlc. Do you get the same issue with vlc?

Although it happens frequently, I don't know how to willingly reproduce it so its hard to say with certainty if problem is ever resolved. I tested out VLC, and it doesnt seem to be as user friendly with Python than omxplayer is. One thing I did as a workaround, I installed pulseaudio to see if that would help out and so far issue hasnt came back

**Comment by popcornmix** [comment]
Assuming you are using the native outputs of omxplayer (local or hdmi) then audio is output from gpu and linux side drivers (like pulseaudio) are not involved.

**Comment by bma-diy** [comment]
Yes I am using hdmi as my output. Yeah I was afraid pulse audio wouldn't be relevant. What I did to pulseaudio config was: edit /etc/pulse/default.pa and commented out "load-module module-suspend-on-idle". I will have to do more testing to see if issue is no longer present. Unfortunatelly I used omxplayer -g and would check /var/log/syslog and i would not see anything obviously jump out to me. It only happened on an insignia Roku TV. I swapped hdmi cable, swapped ports and issue persisted. But on my monitor with speakers attached, issue never came.
